src/FieldWarning/Assets/Ingame/UI/InputManager.cs

```diff
@@ -30,388 +32,327 @@ namespace PFW.Ingame.UI
+                    nextMouseTexture = this._primedReticle;
+                    break;
+                case Modes.Normal:
+                    nextMouseTexture = null;
+                    break;
+                case Modes.ReverseMove:
+                    nextMouseTexture = this._primedReticle;
+                    break;
+                case Modes.Purchasing: break;
+                default: throw new ArgumentOutOfRangeException(nameof(modes), modes, null);
+            }
+
+            Cursor.SetCursor(nextMouseTexture, nextMouseLocation, CursorMode.Auto);
         }
 
-        private void EnterReverseMoveMode()
+        private void ShowGhostUnitsAndMaybePurchase(RaycastHit terrainHover)
         {
-            CurMouseMode = MouseMode.reverseMove;
-            Vector2 hotspot = new Vector2(0, 0);
-            Cursor.SetCursor(_primedReticle, hotspot, CursorMode.Auto);
+            SpawnPointBehaviour closestSpawn = this.GetClosestSpawns(terrainHover.point);
+            this._currentBuyTransaction.PreviewPurchase(
+                terrainHover.point,
+                terrainHover.point * 2 - closestSpawn.transform.position);
+
+            this.MaybePurchaseGhostUnits(closestSpawn);
         }
 
-        public void RegisterPlatoonBirth(PlatoonBehaviour platoon)
+        private void Start()
         {
-            _selectionManager.RegisterPlatoonBirth(platoon);
+            this._firePosReticle = (Texture2D)Resources.Load("FirePosTestTexture");
+            if (this._firePosReticle is null)
+                throw new MissingManifestResourceException("Could not locate Texture2D named 'FirePosTestTexture'.");
+
+            this._primedReticle = (Texture2D)Resources.Load("PrimedCursor");
+            if (this._primedReticle is null)
+                throw new MissingManifestResourceException("Could not locate Texture2D named `PrimedCursor`.");
+
+            this._rightClickManager = new ClickManager(
+                1,
+                this.MoveGhostsToMouse,
+                this.OnOrderHold,
+                this.ExitPurchasingMode,
+                this.MoveGhostsToMouse);
         }
 
-        public void RegisterPlatoonDeath(PlatoonBehaviour platoon)
+        private void Update()
         {
-            _selectionManager.RegisterPlatoonDeath(platoon);
+            this._selectionManager.Update(this.MouseState);
+
+            switch (this.MouseState)
+            {
+                case Modes.Purchasing:
+                    if (Util.GetTerrainClickLocation(out RaycastHit hit)
+                        && hit.transform.gameObject.name.Equals("Terrain"))
+                        this.ShowGhostUnitsAndMaybePurchase(hit);
+                    this.MaybeExitPurchasingMode();
+                    break;
+
+                case Modes.Normal:
+                    this._rightClickManager.Update();
+                    break;
+
+                case Modes.FirePosition:
+                    if (this.IsMouse(Events.Down))
```

Comment:
This takes a method that any Unity developer will recognize and effectively renames it. Not worth it.